test/phoenix_component/components_test.exs

```diff
@@ -19,138 +20,127 @@ defmodule Phoenix.LiveView.ComponentsTest do
-      assert render(~H|<.link href="/">text</.link>|) == ~s|<a href="/">text</a>|
+      assert t2h(~H|<.link href="/">text</.link>|) == ~X|<a href="/">text</a>|
     end
 
     test "arbitrary attrs" do
       assigns = %{}
 
-      assert render(~H|<.link href="/" class="foo">text</.link>|) ==
-               ~s|<a href="/" class="foo">text</a>|
+      assert t2h(~H|<.link href="/" class="foo">text</.link>|) ==
+               ~X|<a href="/" class="foo">text</a>|
     end
 
     test "with no href" do
       assigns = %{}
 
-      assert render(~H|<.link phx-click="click">text</.link>|) ==
-               ~s|<a href="#" phx-click="click">text</a>|
+      assert t2h(~H|<.link phx-click="click">text</.link>|) ==
+               ~X|<a href="#" phx-click="click">text</a>|
     end
 
     test "with # ref" do
       assigns = %{}
 
-      assert render(~H|<.link href="#" phx-click="click">text</.link>|) ==
-               ~s|<a href="#" phx-click="click">text</a>|
+      assert t2h(~H|<.link href="#" phx-click="click">text</.link>|) ==
+               ~X|<a href="#" phx-click="click">text</a>|
     end
 
     test "with nil href" do
       assigns = %{}
 
-      assert render(~H|<.link href={nil} phx-click="click">text</.link>|) ==
-               ~s|<a href="#" phx-click="click">text</a>|
+      assert t2h(~H|<.link href={nil} phx-click="click">text</.link>|) ==
+               ~X|<a href="#" phx-click="click">text</a>|
     end
 
     test "csrf with get method" do
       assigns = %{}
 
-      assert render(~H|<.link href="/" method="get">text</.link>|) ==
-               ~s|<a href="/">text</a>|
+      assert t2h(~H|<.link href="/" method="get">text</.link>|) == ~X|<a href="/">text</a>|
 
-      assert render(~H|<.link href="/" method="get" csrf_token="123">text</.link>|) ==
-               ~s|<a href="/">text</a>|
+      assert t2h(~H|<.link href="/" method="get" csrf_token="123">text</.link>|) ==
+               ~X|<a href="/">text</a>|
     end
 
     test "csrf with non-get method" do
       assigns = %{}
       csrf = Plug.CSRFProtection.get_csrf_token_for("/users")
 
-      assert render(~H|<.link href="/users" method="delete">delete</.link>|) ==
-               ~s|<a href="/users" data-method="delete" data-csrf="#{csrf}" data-to="/users">delete</a>|
+      assert t2h(~H|<.link href="/users" method="delete">delete</.link>|) ==
+               EasyHTML.parse!(
+                 ~s|<a href="/users" data-method="delete" data-csrf="#{csrf}" data-to="/users">delete</a>|
+               )
 
-      assert render(~H|<.link href="/users" method="delete" csrf_token={true}>delete</.link>|) ==
-               ~s|<a href="/users" data-method="delete" data-csrf="#{csrf}" data-to="/users">delete</a>|
+      assert t2h(~H|<.link href="/users" method="delete" csrf_token={true}>delete</.link>|) ==
+               EasyHTML.parse!(
+                 ~s|<a href="/users" data-method="delete" data-csrf="#{csrf}" data-to="/users">delete</a>|
+               )
```

Comment:
I couldn't create the lowercase sigil to interpolate, so I used the function directly in this case. Maybe it is possible to do it but I don't know how.